File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/GetFileAttachmentsUseCase.java
package com.glia.widgets.core.fileupload.domain;

import com.glia.widgets.core.fileupload.FileAttachmentRepository;
import com.glia.widgets.core.fileupload.model.FileAttachment;

import java.util.List;

public class GetFileAttachmentsUseCase {
    private final FileAttachmentRepository repository;

    public GetFileAttachmentsUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public List<FileAttachment> execute() {
        return repository.getFileAttachments();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/GetFileAttachmentsUseCase.kt
package com.glia.widgets.core.fileupload.domain

import com.glia.widgets.core.fileupload.FileAttachmentRepository
import com.glia.widgets.core.fileupload.model.LocalAttachment

internal class GetFileAttachmentsUseCase(private val repository: FileAttachmentRepository) {
    operator fun invoke(): List<LocalAttachment> {
        return repository.getFileAttachments()
    }
}
